cmd/osmosisd/cmd/init.go

```diff
@@ -136,53 +141,82 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 					return errors.New("invalid mnemonic")
 				}
 			}
-
 			nodeID, _, err := genutil.InitializeNodeValidatorFilesFromMnemonic(config, mnemonic)
 			if err != nil {
 				return err
 			}
 
-			config.Moniker = args[0]
-
-			genFile := config.GenesisFile()
+			genFilePath := config.GenesisFile()
+			chainID, _ := cmd.Flags().GetString(flags.FlagChainID)
 			overwrite, _ := cmd.Flags().GetBool(FlagOverwrite)
 
-			if !overwrite && tmos.FileExists(genFile) {
-				return fmt.Errorf("genesis.json file already exists: %v", genFile)
-			}
-			appState, err := json.MarshalIndent(mbm.DefaultGenesis(cdc), "", " ")
-			if err != nil {
-				return errors.Wrap(err, "Failed to marshall default genesis state")
+			if !overwrite && tmos.FileExists(genFilePath) {
+				return fmt.Errorf("genesis.json file already exists: %v", genFilePath)
 			}
 
-			genDoc := &types.GenesisDoc{}
-			if _, err := os.Stat(genFile); err != nil {
-				if !os.IsNotExist(err) {
-					return err
-				}
-			} else {
-				genDoc, err = types.GenesisDocFromFile(genFile)
+			var toPrint printInfo
+			isMainnet := chainID == "" || chainID == "osmosis-1"
+			genesisFileDownloadFailed := false
+
+			if isMainnet {
+				// If the chainID is blank or osmosis-1, prep this as a mainnet node
+
+				// Attempt to download the genesis file from the Osmosis GitHub repository
+				// If fail, generate a new genesis file
+				err := downloadGenesis(config)
 				if err != nil {
-					return errors.Wrap(err, "Failed to read genesis doc from file")
+					fmt.Println("Failed to download genesis file, using default")
```

Comment:
```suggestion
					fmt.Println("Failed to download genesis file, using a random chain ID and genesis file for local testing")
```

Reply:
Lets add a TODO indicating that this is a weird thing to do here, and maybe we should just fail?

Reply:
This is actually the normal behavior of what happens right now, so its essentially defaulting to what it is doing now

Reply:
But yeah happy to add a note on this